kod/object/passive/spell.kod

```diff
@@ -481,6 +481,24 @@ messages:
       return TRUE;
    }
 
+   GetRequiredKarma()
+   "Returns the karma value required to use the spell, or 0 if there's no karma requirement."
+   {
+      local iKarma;
+
+      iKarma = 0;
+      if (viSchool = SS_QOR)
```

Comment:
This is pretty much a duplicate of the code in KarmaCheck.  It should exist only in one place.  I think you can change KarmaCheck to call this function.

Reply:
KarmaCheck checks the spell's karma against the player's karma and returns whether or not they can cast it with their current karma.  I thought about using this approach, but doing so would only show a karma message in the descriptions if the player lacked the karma to use the item, and only if they were holding it.

The two functions are similar though, and while I also thought about adding a call to it in KarmaCheck (so that the if school is qor or shal'ille check only existed in one place) but felt that went outside the scope of the PR.

I'm open to recommendations on all this.

Reply:
KarmaCheck updated. :)